src/Graph/GraphFormatter.php

```diff
@@ -0,0 +1,227 @@
+<?php
+
+namespace SRF\Graph;
+
+use Html;
+
+/**
+ *
+ *
+ * @see https://www.semantic-mediawiki.org/wiki/Help:Graph_format
+ *
+ * @license GNU GPL v2+
+ * @since 3.2
+ *
+ * @author Sebastian Schmid (gesinn.it)
+ *
+ */
+
+class GraphFormatter {
+
+	private $graph = "";
+
+	protected $graphColors = [
+		'black',
+		'red',
+		'green',
+		'blue',
+		'darkviolet',
+		'gold',
+		'deeppink',
+		'brown',
+		'bisque',
+		'darkgreen',
+		'yellow',
+		'darkblue',
+		'magenta',
+		'steelblue2'
+	];
+	private $legendItem = [];
+	private $options = [];
+
+	public function __construct( $options ){
```

Comment:
An object to hold the options rather than an array has the big advantage that you can use code analysis (things like auto completion, automated refactoring and type mismatch detection)

Reply:
@see: https://github.com/RacoonDev/SemanticResultFormats/commit/ade773b60f5bdb9dabfe95d21b30251298898c90

Reply:
Huh.

Sorry it seems I was not clear enough. My concern is not about object vs array SYNTAX. It is about free-form-dynamic-whatever vs defined-structure. In other words, I was thinking about a class like GraphFormatterOptions with fields similar to those that where present in the class you are extracting from.

Reply:
Just a bunch of public fields in this class will already be a big improvement. A next step could be adding type docs to the fields.

Reply:
Okey I will create this class but I think "GraphOptions" would be a better name for it. Should I just use public fields or should I declare them private and implement some getter functions?

Reply:
PHP does not have a nice way to create this type of objects... you can read my rant on it here https://www.entropywins.wtf/blog/2016/02/03/missing-in-php7-value-objects/

Since this `GraphOptions` is used only locally I'd be tempted with just having public fields and saving myself the hassle of adding extra stuff.

Reply:
I created now the GraphOptions Class with its public fields and the related getter functions. For the initialising i pass the params-array of the SMWResultPrinter to the GraphOptions constructor and map the public fields based on the array keys.

Reply:
i see you went with private fields and getters. This works quite nicely since you need to convert from an array anyway, meaning your constructor does not end up with 1000 parameters. This is nicer than what I was thinking about. You now also have return types \o/ That means the type docs for the fields are pretty superfluous and can be removed.